vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/gemini/post_account/post_account_gemini_unittest.cc

```diff
@@ -0,0 +1,133 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <memory>
+#include <string>
+
+#include "base/test/task_environment.h"
+#include "bat/ledger/internal/endpoint/gemini/post_account/post_account_gemini.h"
+#include "bat/ledger/internal/ledger_client_mock.h"
+#include "bat/ledger/internal/ledger_impl_mock.h"
+#include "bat/ledger/ledger.h"
+#include "net/http/http_status_code.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+// npm run test -- brave_unit_tests --filter=PostAccountGeminiTest.*
+
+using ::testing::_;
+using ::testing::Invoke;
+
+namespace ledger {
+namespace endpoint {
+namespace gemini {
+
+class GeminiPostAccountTest : public testing::Test {
+ private:
+  base::test::TaskEnvironment scoped_task_environment_;
+
+ protected:
+  std::unique_ptr<ledger::MockLedgerClient> mock_ledger_client_;
+  std::unique_ptr<ledger::MockLedgerImpl> mock_ledger_impl_;
+  std::unique_ptr<PostAccount> post_account_;
+
+  GeminiPostAccountTest() {
+    mock_ledger_client_ = std::make_unique<ledger::MockLedgerClient>();
+    mock_ledger_impl_ =
+        std::make_unique<ledger::MockLedgerImpl>(mock_ledger_client_.get());
+    post_account_ = std::make_unique<PostAccount>(mock_ledger_impl_.get());
+  }
+};
+
+TEST_F(GeminiPostAccountTest, ServerOK) {
+  ON_CALL(*mock_ledger_client_, LoadURL(_, _))
+      .WillByDefault(Invoke(
+          [](type::UrlRequestPtr request, client::LoadURLCallback callback) {
+            type::UrlResponse response;
+            response.status_code = 200;
+            response.url = request->url;
+            response.body = R"({
+              "account": {
+                "accountName": "Primary",
+                "shortName": "primary",
+                "type": "exchange",
+                "created": "1619040615242",
+                "verificationToken": "mocktoken"
+              },
+              "users": [{
+                "name": "Test",
+                "lastSignIn": "2021-04-30T18:46:03.017Z",
+                "status": "Active",
+                "countryCode": "US",
+                "isVerified": true
+              }],
+              "memo_reference_code": "GEMAPLLV"
+            })";
+            callback(response);
+          }));
+
+  post_account_->Request(
+      "4c2b665ca060d912fec5c735c734859a06118cc8",
+      [](const type::Result result, const std::string address,
+         const std::string linking_info, const std::string user_name,
+         const bool verified) {
+        EXPECT_EQ(result, type::Result::LEDGER_OK);
+        EXPECT_EQ(address, "Primary");
+        EXPECT_EQ(linking_info, "mocktoken");
+        EXPECT_EQ(user_name, "Test");
+        EXPECT_TRUE(verified);
+      });
+}
+
+TEST_F(GeminiPostAccountTest, ServerError401) {
+  ON_CALL(*mock_ledger_client_, LoadURL(_, _))
+      .WillByDefault(Invoke(
+          [](type::UrlRequestPtr request, client::LoadURLCallback callback) {
+            type::UrlResponse response;
+            response.status_code = 401;
```

Comment:
`net::HTTP_UNAUTHORIZED` instead of hardcoded value

Reply:
resolved